### PR TITLE
[Feature][JDBC]Support Split By Date Type Column

### PR DESCRIPTION
**Pull request by EricJoy2048**
### Purpose of this pull request

Add split by date type column.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by hailin0 — DISMISSED** [review]

**Comment by hailin0** [comment]
It is recommended to use the fork warehouse to submit code next time

**Comment by EricJoy2048** [comment]
> It is recommended to use the fork warehouse to submit code next time

Ok.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/DynamicChunkSplitter.java [hunk 1]
```diff
@@ -309,6 +329,71 @@ private List<ChunkRange> splitUnevenlySizedChunks(
...
+            sqlDateMin = (Date) min;
+            sqlDateMax = (Date) max;
+        } else if (min instanceof Timestamp) {
+            sqlDateMin = new Date(((Timestamp) min).getTime());
+            sqlDateMax = new Date(((Timestamp) max).getTime());
+        }
+        List<LocalDate> dateRange =
+                getDateRange(sqlDateMin.toLocalDate(), sqlDateMax.toLocalDate());
+        if (dateRange.size() > 20 * 365) {
+            // TODO: If dateRange granter than 20 year, need get the real date in the table
```
Comment (zhilinli123) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/DynamicChunkSplitter.java:361]: Can the todo logic here be reflected in the document? This is also convenient for user perception

File: docs/en/connector-v2/source/Jdbc.md [hunk 2]
```diff
@@ -239,29 +262,61 @@ sink {
 }
 ```
 
-Using `table_path` read:
+### Parallel Boundary:
 
-***Configuring `table_path` will turn on auto split, you can configure `split.*` to adjust the split strategy***
+> It is more efficient to specify the data within the upper and lower bounds of the query It is more efficient to read your data source according to the upper and lower boundaries you configured
```
Comment (TaoZex) [inline docs/en/connector-v2/source/Jdbc.md]:
```suggestion
> It is more efficient to specify the data within the upper and lower bounds of the query. It is more efficient to read your data source according to the upper and lower boundaries you configured.
```

**Review by TaoZex — APPROVED** [review]
LGTM. Only a small typo need to fix.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/DynamicChunkSplitter.java (hunk 1 above)
Comment (EricJoy2048) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/DynamicChunkSplitter.java:361]: > Can the todo logic here be reflected in the document? This is also convenient for user perception

In fact, each type of split will be calculated differently in different situations. At present, we do not have a special document for this section. In the future, we need a special introduction document for split algorithm, and we need to discuss what directory it should be placed in through the mailing list.

**Review by hailin0 — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java [hunk 3]
```diff
@@ -40,6 +40,7 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.logging.Logger;
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java]: Please use `org.slf4j.Logger;`

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java [hunk 4]
```diff
@@ -50,6 +51,8 @@
  */
 public interface JdbcDialect extends Serializable {
 
+    Logger log = Logger.getLogger(JdbcDialect.class.getName());
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java]:
```suggestion
    private static Logger log = Logger.getLogger(JdbcDialect.class.getName());
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-7/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcMysqlSplitIT.java [hunk 5]
```diff
@@ -0,0 +1,500 @@
...
+
+        return Pair.of(fieldNames, rows);
+    }
+
+    static JdbcUrlUtil.UrlInfo mysqlUrlInfo =
+            JdbcUrlUtil.getUrlInfo(
+                    String.format("jdbc:mysql://localhost:%s/auto?useSSL=false", MYSQL_PORT));
+
+    @TestTemplate
+    public void testSplit(TestContainer container) throws Exception, InterruptedException {
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-7/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcMysqlSplitIT.java]:
```suggestion
    @Test
    public void testSplit() {
```
Please use `@Test` not `@TestTemplate`

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-7/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcMysqlSplitIT.java [hunk 6]
```diff
@@ -0,0 +1,500 @@
...
+                        .build();
+        Collection<JdbcSourceSplit> jdbcSourceSplits = splitter.generateSplits(jdbcSourceTable);
+
+        Assertions.assertEquals(11, jdbcSourceSplits.size());
+
+        // use tinyint column to split
+        configMap.put("partition_column", "c_tinyint");
+        splitter = getDynamicChunkSplitter(configMap);
+        jdbcSourceSplits = splitter.generateSplits(jdbcSourceTable);
+        Assertions.assertEquals(11, jdbcSourceSplits.size());
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-7/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcMysqlSplitIT.java]: Please check each split range too.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java (hunk 4 above)
Comment (EricJoy2048) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java]: done

**Review by Hisoka-X — APPROVED** [review]

**Review by ic4y — APPROVED** [review]

**Pull request merged**